pubky-homeserver/src/persistence/lmdb/tables/users.rs

```
@@ -80,6 +80,42 @@ pub enum UserQueryError {
 }
 
 impl LmDB {
+    /// Updates a user's data usage by a signed `delta` (bytes).
+    /// Increases or decreases the stored `used_bytes` count for `public_key`.
+    /// Negative results are clamped to zero to prevent underflow.
+    pub fn update_data_usage(&mut self, public_key: &PublicKey, delta: i64) -> anyhow::Result<()> {
+        let mut wtxn = self.env.write_txn()?;
+        let mut user = self
+            .tables
+            .users
+            .get(&wtxn, public_key)?
+            .unwrap_or(User::default());
```

If the pubky does not belong to one of our users, should a new user be created? It should throw an error, no?

---

Yes, correct. Silently creating a brand-new User::default() on a typo’d or unknown key can mask bugs. Replacing it for.
```rust
        .ok_or_else(|| anyhow::anyhow!("no user found for public key {:?}", public_key))?;
```